modules/fedimint-lnv2-client/src/lib.rs

```diff
@@ -0,0 +1,725 @@
+    ) -> Option<TransactionItemAmount> {
+        Some(TransactionItemAmount {
+            amount: input.amount,
+            fee: self.cfg.fee_consensus.contract_input,
+        })
+    }
+
+    fn output_amount(
+        &self,
+        output: &<Self::Common as ModuleCommon>::Output,
+    ) -> Option<TransactionItemAmount> {
+        Some(TransactionItemAmount {
+            amount: output.amount(),
+            fee: self.cfg.fee_consensus.contract_output,
+        })
+    }
+}
+
+fn generate_ephemeral_tweak(static_pk: PublicKey) -> ([u8; 32], PublicKey) {
+    let ephemeral_keypair = KeyPair::new(secp256k1::SECP256K1, &mut rand::thread_rng());
+
+    let ephemeral_tweak = ecdh::shared_secret_point(&static_pk, &ephemeral_keypair.secret_key())
+        .consensus_hash::<sha256::Hash>()
+        .into_inner();
+
+    (ephemeral_tweak, ephemeral_keypair.public_key())
+}
+
+impl LightningClientModule {
+    pub async fn payment_info(&self, gateway_api: SafeUrl) -> anyhow::Result<PaymentInfo> {
+        reqwest::Client::new()
+            .post(
+                gateway_api
+                    .join("payment_info")
+                    .expect("'payment_info' contains no invalid characters for a URL")
+                    .as_str(),
+            )
+            .json(&self.federation_id)
+            .send()
+            .await?
+            .json::<Option<PaymentInfo>>()
+            .await?
+            .ok_or(anyhow!("The gateway does not support our federation"))
+    }
+
+    pub async fn send(
+        &self,
+        gateway_api: SafeUrl,
+        invoice: Bolt11Invoice,
+    ) -> anyhow::Result<OperationId> {
+        self.send_internal(
+            gateway_api,
+            invoice,
+            PaymentFee::default(),
+            SEND_EXPIRATION_DELTA_BLOCKS_DEFAULT,
+        )
+        .await
+    }
+
+    pub async fn send_internal(
+        &self,
+        gateway_api: SafeUrl,
+        invoice: Bolt11Invoice,
+        max_fee: PaymentFee,
+        expiration_delta: u64,
+    ) -> anyhow::Result<OperationId> {
+        let (ephemeral_tweak, ephemeral_pk) = generate_ephemeral_tweak(self.keypair.public_key());
+
+        let refund_keypair = SecretKey::from_slice(&ephemeral_tweak)
+            .expect("32 bytes, within curve order")
+            .keypair(secp256k1::SECP256K1);
+
+        let payment_info = self
+            .payment_info(gateway_api.clone())
+            .await
+            .context("Gateway is offline")?;
+
+        ensure!(
+            max_fee >= payment_info.payment_fees.send,
+            "The gateway send fee is above {max_fee:?}"
+        );
+
+        // we double it to account for cltv expiry deltas along the lightning route
+        let min_outgoing_cltv_delta = 2 * payment_info.outgoing_cltv_delta;
+
+        ensure!(
+            min_outgoing_cltv_delta <= expiration_delta,
+            "The minimum possible expiration delta is {min_outgoing_cltv_delta} blocks"
```

Comment:
Do we need to fail here? Couldn't we just use `expiration_delta` if we deem `min_outgoing_cltv_delta` is too low? Would also help to modify the error message to indicate this is a gateway LN node config issue, since clients could hit this and wonder why their payments are not working.

Reply:
Ok sorry I think I got confused earlier, we need to fail here since its not safe to use a cltv_delta that is lower than the gateway's. A better error message would be helpful here